x-pack/plugins/observability_solution/observability/public/pages/alert_details/components/alert_history.tsx

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { ALERTING_FEATURE_ID } from '@kbn/alerting-plugin/common';
+import {
+  EuiPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiText,
+  EuiSpacer,
+  EuiLoadingSpinner,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { ALERT_INSTANCE_ID, ALERT_RULE_UUID, type AlertConsumers } from '@kbn/rule-data-utils';
+import { useAlertsHistory } from '@kbn/observability-alert-details';
+import { convertTo } from '../../../../common/utils/formatters';
+import { useFetchRuleTypes } from '../../../hooks/use_fetch_rule_types';
+import { useGetFilteredRuleTypes } from '../../../hooks/use_get_filtered_rule_types';
+import { useKibana } from '../../../utils/kibana_react';
+import { getDefaultAlertSummaryTimeRange } from '../../../utils/alert_summary_widget';
+import {
+  CustomThresholdAlert,
+  CustomThresholdRule,
+} from '../../../components/custom_threshold/components/types';
+
+interface Props {
+  alert: CustomThresholdAlert;
+  rule: CustomThresholdRule;
```

Comment:
The rule here couldn't be any rule type from this list Log, APM latency, or SLO?

Reply:
Good catch, I will update it.

Reply:
Fixed in [6d29e94](https://github.com/elastic/kibana/pull/181824/commits/6d29e941831f98cf59cd5c239327bc74c9e11164)